plugin/pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -52,19 +51,48 @@ func NewInterPodAffinityPriority(
 	return interPodAffinity.CalculateInterPodAffinityPriority
 }
 
-// TODO: Share it with predicates by moving to better location.
-// TODO: Can we avoid error handling here - this is only a matter of non-parsable selector?
-func podMatchesNamespaceAndSelector(pod *api.Pod, affinityPod *api.Pod, term *api.PodAffinityTerm) (bool, error) {
-	namespaces := priorityutil.GetNamespacesFromPodAffinityTerm(affinityPod, *term)
-	if len(namespaces) != 0 && !namespaces.Has(pod.Namespace) {
-		return false, nil
+type podAffinityPriorityMap struct {
+	// nodes contain all nodes that should be considered
+	nodes []*api.Node
+	// counts store the mapping from node name to so-far computed score of
+	// the node.
+	counts map[string]float64
+	// failureDomains contain default failure domains keys
+	failureDomains priorityutil.Topologies
+	// The first error that we faced.
+	firstError error
+}
+
+func newPodAffinityPriorityMap(nodes []*api.Node, failureDomains priorityutil.Topologies) *podAffinityPriorityMap {
+	return &podAffinityPriorityMap{
+		nodes:          nodes,
+		counts:         make(map[string]float64, len(nodes)),
+		failureDomains: failureDomains,
 	}
+}
 
-	selector, err := unversioned.LabelSelectorAsSelector(term.LabelSelector)
-	if err != nil || !selector.Matches(labels.Set(pod.Labels)) {
-		return false, err
+func (p *podAffinityPriorityMap) processTerm(term *api.PodAffinityTerm, podDefiningAffinityTerm, podToCheck *api.Pod, fixedNode *api.Node, weight float64) {
+	match, err := priorityutil.PodMatchesTermsNamespaceAndSelector(podToCheck, podDefiningAffinityTerm, term)
+	if err != nil {
+		if p.firstError == nil {
+			p.firstError = err
+		}
+		return
+	}
+	if match {
+		for _, node := range p.nodes {
+			if p.failureDomains.NodesHaveSameTopologyKey(node, fixedNode, term.TopologyKey) {
+				p.counts[node.Name] += weight
+			}
+		}
+	}
+}
+
+func (p *podAffinityPriorityMap) processTerms(terms []api.WeightedPodAffinityTerm, podDefiningAffinityTerm, podToCheck *api.Pod, fixedNode *api.Node, multiplier int) {
```

Comment:
FYI Those two methods are basically a copy from what was previously an anonymous function in the code.